Give users the option of disabling auto-frame-track

karupayun opened the pull request:
In this PR we are including an option to disable auto frame track in
user settings (by default true) as it was discussed in
go/stadia-orbit-auto-frame-track-alignment. The option is related
about Orbit adding automatically the default Frame Track at the start
of the session.

The logic is a bit more complex as it seems.
 - The option is true by default.
 - Setting the flag to false won't erase the already added default frame
   track. It will apply for the next session.
 - Setting the flag to true will add the frame track during the next
   LoadAllSymbols call. This will be modified in a future commit, I
   think we should add it as soon as possible (by actually triggering
   AddDefaultFrameTrack method) but only once (so users are able to
   erase it manually: TODO tracked in http://b/239675491), but I'm open
   to suggestions if you think that we should only add it in the next
   session to be consistent.

http://screenshot/BYU8NuWCCr3h93u

Bug: http://b/239148938
Test: Started/ended sessions flipping the option and took several
captures. Modified E2E tests (https://github.com/google/orbit/pull/3998)
and run them manually. In progress: Creating a signed build and triggering 
E2E tests.

reichlfl commented:
I didn't look at the code in detail, but one comment regarding the screenshot: It could be confusing to users to have this option that refers to the whole session amidst the other options that refer to the next capture. Especially since the text next to the "OK" Button states that the options will come into effect with the next capture, and I don't think the tooltip helps much.

At the very least I'd make the tooltip more explicit that this is a session-wide setting. The way I read the tooltip is that if I enable this setting at the beginning of a session, a frametrack will be added. But I think it's only added if the setting has already been active at the beginning of a session, i.e. before symbol loading, is that correct? That feels confusing.

Did you consider adding the option in a separate place in the options Window to make this clearer? You could e.g. add a toolbox or tab widget with 2 pages, one for the "Next Capture" options and one for the "Next Session" options, and add the text explaining when these settings come into effect within those pages instead of next to the "OK" button.

karupayun commented:
> At the very least I'd make the tooltip more explicit that this is a session-wide setting. The way I read the tooltip is that if I enable this setting at the beginning of a session, a frametrack will be added. But I think it's only added if the setting has already been active at the beginning of a session, i.e. before symbol loading, is that correct? That feels confusing.

A few things. The message and the tooltip was agreed with Ronald before the creation of this PR. The motivation was making it clear that if the user erase the automatic FrameTrack manually, it won't appear in the following captures, so basically that it will be added only once during a session (which is currently a TODO tracked in http://b/239675491). 

> The way I read the tooltip is that if I enable this setting at the beginning of a session, a frametrack will be added. But I think it's only added if the setting has already been active at the beginning of a session, i.e. before symbol loading, is that correct? That feels confusing.

No, is what you got from the tooltip. The already added FrameTrack won't be erased (because the complexity related to erasing a FrameTrack) but if the user is choosing the option to be true again, I will add it as soon as possible, because I think it makes sense (TODO).

> Did you consider adding the option in a separate place in the options Window to make this clearer? You could e.g. add a toolbox or tab widget with 2 pages, one for the "Next Capture" options and one for the "Next Session" options, and add the text explaining when these settings come into effect within those pages instead of next to the "OK" button.

No, I found it very complex. We are just speaking about a border case-I don't expect the user to select/deselect this option very often. I will simply clarify using a better message making it as clear as possible the current behavior. My plan, discussed with Ronald, is to insert it ASAP but not remove the already added FrameTrack. . And also only add it once (at the start of the session OR when the user enable it). I could add "the Frame Track won't be removed for this session", what do you think?

ronaldfw commented on src/OrbitQt/CaptureOptionsDialog.ui:
```diff
@@ -405,6 +405,28 @@ p, li { white-space: pre-wrap; }
+             <string>AutoFrameTrackCheckBox</string>
+            </property>
+            <property name="text">
+             <string>Automatically add the default frame track</string>
```
Using "the" in "the default frame track" sounds a bit strange to me, because there is not a canonically and uniquely defined frame track (there are choices in the function used, and whether to base it on start or end of the timeslice, etc.)

Suggestion:
"Automatically add a render frame track by default."

karupayun replied:
Added the suggestion. Also the missing ⓘ that indicates that there is a tooltip.

ronaldfw reviewed (approved):
LGTM

Minor suggestion on wording.

karupayun commented:
> > but if the user is choosing the option to be true again, I will add it as soon as possible
> 
> Ok, that makes sense. Then I find the option less confusing than I initially thought. Thanks for clarifying, then I'm fine with those changes. Maybe as you suggested add something like "Disabling this option will not remove an existing frame track for the current session" or so.

Addressed Ronald's comment and added "Disabling this option will not remove *any* existing frame track for the current session" here.

Henning gave me ok in merging it without his approval, so I will merge this in the next hour.
